src/process.c

```diff
@@ -562,18 +654,18 @@ cc_shim_launch (struct cc_oci_config *config,
 			goto child_failed;
 		}
 
-		/* +1 for for NULL terminator */
+		/* remove FD_CLOEXEC flags */
+		flags = fcntl(proxy_socket_fd, F_GETFD);
+		flags &= ~FD_CLOEXEC;
+		fcntl(proxy_socket_fd, F_SETFD, flags);
```

Comment:
I'd be tempted to rework `cc_oci_fd_set_cloexec()` to be:

```
gboolean
cc_oci_fd_toggle_cloexec (int fd, gboolean set)
```

so that you can call:

```
cc_oci_fd_toggle_cloexec (proxy_socket_fd, false);
cc_oci_fd_toggle_cloexec (proxy_io_fd, false);
```